RELEASING.md

```diff
@@ -12,76 +65,143 @@ export NEW_RELEASE=0.4.0.dev4
 
-This step is only needed, if you want to include a summary text in the autogenerated changelog.
+### Create $NEW_RELEASE milestone in github
 
-* Create a new issue and label it with release-summary
-* Create $NEW_RELEASE milestone in github, and assign the issue to that
-* Close the issue
+If not already created
 
-3. Generate changelog
+### Create new issue linked to the milestone
 
 ```bash
-# gem install github_changelog_generator --pre
-# https://github.com/github-changelog-generator/github-changelog-generator#github-token
-export CHANGELOG_GITHUB_TOKEN=token
-github_changelog_generator --base HISTORY.md --user python-kasa --project python-kasa --since-tag $PREVIOUS_RELEASE --future-release $NEW_RELEASE -o CHANGELOG.md
+gh issue create --label "release-summary" --milestone $NEW_RELEASE --title "$NEW_RELEASE Release Summary" --body "Some summary text"
+```
+
+You can exclude the --body option to get an interactive editor or leave blank and go into the issue on github and edit there.
+
+### Close the issue
+
+Either via github or:
+
+```bash
+gh issue close ISSUE_NUMBER
+```
+
+## Generate changelog
+
+### For pre-release
 
+EXCLUDE_TAGS will exclude all dev tags except for the current release dev tags.
 
-#-exclude-tags-regex 'dev\d$'
+Regex should be something like this `^((?!0\.7\.0)(.*dev\d))+`. The first match group negative matches on the current release and the second matches on releases ending with dev.
+
+```bash
+EXCLUDE_TAGS=${NEW_RELEASE%.dev*}; EXCLUDE_TAGS=${EXCLUDE_TAGS//"."/"\."}; EXCLUDE_TAGS="^((?!"$EXCLUDE_TAGS")(.*dev\d))+"
+github_changelog_generator --base HISTORY.md --user python-kasa --project python-kasa --since-tag $PREVIOUS_RELEASE --future-release $NEW_RELEASE -o CHANGELOG.md --exclude-tags-regex "$EXCLUDE_TAGS"
+```
+
+### For production
+
+```bash
+github_changelog_generator --base HISTORY.md --user python-kasa --project python-kasa --since-tag $PREVIOUS_RELEASE --future-release $NEW_RELEASE -o CHANGELOG.md --exclude-tags-regex 'dev\d$'
 ```
 
-Remove '--exclude-tags-regex' for dev releases.
-TODO: Find a nice way to collate dev* changes to real releases. Excluding them causes them to re-appear for pre-releases, but maybe that's fine?
+You can ignore warnings about missing PR commits like below as these relate to PRs to branches other than master:
+```
+Warning: PR 908 merge commit was not found in the release branch or tagged git history and no rebased SHA comment was found
+```
 
-## Commit the changed files
+
+## Export new release notes to variable
 
 ```bash
-git commit --all --verbose
+export RELEASE_NOTES=$(grep -Poz '(?<=\# Changelog\n\n)(.|\n)+?(?=\#\#)' CHANGELOG.md | tr '\0' '\n' )
+echo "$RELEASE_NOTES"
+```
+
+Check the release notes output and if it doesn't look right or the command doesn't work just copy paste from CHANGELOG.md for the latest release:
+
+```bash
+export RELEASE_NOTES="paste release notes from CHANGELOG.md here"
+```
```

Comment:
```suggestion
```
This is so obvious that we don't need to state it for ourselves :-)

Reply:
I don't know how cross-platform the regex is so it thought it prudent to stress to check.  Maybe I'll just add it as a comment to the `echo` bash line.

Reply:
Yeah, maybe it's a good idea to print out the variables prior doing steps that change something?

Reply:
Removed and comment added to the `echo` bash line